### PR TITLE
Support tags with a minimal change

### PR DESCRIPTION
**Pull request by bom-d-van**
* Add a simple support for tags for line protocol
* Drops string pointer from normalizePath
* Cherry-pick changes from https://github.com/bookingcom/nanotube/pull/29 by @jwkohnen (Couldn't rebase my works on jwkohnen's change due to conflicts)

(I do agree with @dgryski on using byte array directly returned from net.Conn, but that's bigger change.)

Adding this simple tag support for exploring a new solution of supporting high cardinality metrics in go-carbon.

### CODE REVIEW

**Comment by grzkv** [comment]
Thanks for the PR!
We are currently doing some fixing in core logic and will address this PR after we are done. It may happen that you will have to do a bit of merging. Sorry about that.

**Comment by grzkv** [comment]
Could you please add the benchmark results to the final review? It makes sense to do that after resolving the conflicts.

File: pkg/rec/rec_test.go [hunk 1]
```diff
@@ -165,60 +176,69 @@ func TestRec(t *testing.T) {
...
+			path: "ab___cd.a__b._zkl",
 		},
 		{
-			in:  "ab^%+=.cdef.jk&",
-			out: "ab____.cdef.jk_",
+			in:   "ab^%+=.cdef.jk&",
+			path: "ab____.cdef.jk_",
+		},
+		{
+			in:   "ab^%+=.cdef.jk&;tag1=val1",
```
Comment (grzkv) [inline pkg/rec/rec_test.go]: Could you please add a couple more tests. It would be nice if they contained edge cases.

**Comment by grzkv** [comment]
>  We are currently doing some fixing in core logic and will address this PR after we are done. It may happen that you will have to do a bit of merging. Sorry about that.

We are done with the mentioned fixes.

**Comment by grzkv** [comment]
@bom-d-van @jwkohnen Could you please share your plan on this PR? Thanks!

**Comment by bom-d-van** [comment]
Hi @grzkv, thanks for pinging. if this blocking you, feel free to move ahead (or take over the changes if you would like to add tags support). it might take me some time come back to this topic.

**Comment by grzkv** [comment]
@bom-d-van Thanks for the update. There are several points:
* We are not planning to work on tags at this moment but maybe in the near future. We are not ready to overtake it.
* The `master` went forward quite a bit. 
* It is likely that there are some significant changes to the logic needed in the current implementation.
* This PR includes the #29 that will be merged and rolled out separately.
* This PR partially includes #61 that will be worked on separately. The work will start now. It will conflict.

According to these points, I think it makes sense to close this PR and resume the work with a new one.  You can do this when you see fit by manually moving the changes.
